### PR TITLE
LW-9418 Add Conway era txs e2e tests

### PR DESCRIPTION
**Pull request by iccicci**
# Context

We need to cover Conway era transactions in our e2e test.

# Proposed Solution

Added a test suite with many (hopefully all) transactions with new features in Conway era

# Important Changes Introduced

- Fixed DRep representation in related certificates
- Fixed newly added deposits in Conway era in `DbSyncChainHistoryProvider`
- Fixed `NewConstitution` proposal representation in `DbSyncChainHistoryProvider`
- Bumped protocol major version in local-testnet
- Fixed containers dependencies in docker compose infrastructure

### CODE REVIEW

File: packages/cardano-services-client/src/TxSubmitProvider/TxSubmitApiProvider.ts [hunk 1]
```diff
@@ -43,7 +43,11 @@ export class TxSubmitApiProvider implements TxSubmitProvider {
 
         if (typeof status === 'number' && status >= 400 && status < 500) this.#healthStatus = true;
 
-        throw new ProviderError(ProviderFailure.BadRequest, mapCardanoTxSubmitError(data), data as string);
+        throw new ProviderError(
+          ProviderFailure.BadRequest,
+          mapCardanoTxSubmitError(data),
+          typeof data === 'string' ? data : JSON.stringify(data)
```
Comment (mirceahasegan) [inline packages/cardano-services-client/src/TxSubmitProvider/TxSubmitApiProvider.ts:49]: I think JSON.stringify will work for simple strings too:
```diff
-          typeof data === 'string' ? data : JSON.stringify(data)
+          typeof data === 'string' ? data : JSON.stringify(data)
```

File: packages/e2e/test/wallet/PersonalWallet/conwayTransactions.test.ts [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    );
+    const dRepRegCert: Cardano.RegisterDelegateRepresentativeCertificate = {
+      __typename: Cardano.CertificateType.RegisterDelegateRepresentative,
+      anchor,
+      dRepCredential: { hash: dRepKeyHash, type: Cardano.CredentialType.KeyHash },
+      deposit: dRepDeposit
+    };
+
+    // eslint-disable-next-line no-console
+    console.log(dRepRegCert);
```
Comment (mirceahasegan) [inline packages/e2e/test/wallet/PersonalWallet/conwayTransactions.test.ts:165]: TODO: remove before merging

File: packages/e2e/test/wallet/PersonalWallet/conwayTransactions.test.ts [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+    const txFoundInHistory = await firstValueFromTimed(
+      wallet.transactions.history$.pipe(
+        map((txs) => txs.find((tx) => tx.id === signedTx.tx.id)),
+        filter(isNotNil),
+        take(1)
+      )
+    );
+
+    expect(txFoundInHistory.body.certificates?.length).toBe(1);
+    expect(txFoundInHistory.body.certificates![0]).toEqual(dRepRegCert);
```
Comment (mirceahasegan) [inline packages/e2e/test/wallet/PersonalWallet/conwayTransactions.test.ts:141]:
```diff
-    expect(txFoundInHistory.body.certificates![0]).toEqual(dRepRegCert);
+     assertTxHasCertificate...

File: packages/cardano-services-client/src/TxSubmitProvider/TxSubmitApiProvider.ts (hunk 1 above)
Comment (iccicci) [reply packages/cardano-services-client/src/TxSubmitProvider/TxSubmitApiProvider.ts:49]: Yes it works, but it also encodes the string: it adds starting and ending double quotes and escapes newlines and double quotes.
I think it's better avoiding one more encoding level if not necessary.

File: packages/e2e/test/wallet/PersonalWallet/conwayTransactions.test.ts (hunk 3 above)
Comment (iccicci) [reply packages/e2e/test/wallet/PersonalWallet/conwayTransactions.test.ts:141]: I've not worked yet on this test... it needs to be completely reworked

File: packages/core/src/Serialization/TransactionBody/ProposalProcedure/InfoAction.ts [hunk 4]
```diff
@@ -22,6 +24,7 @@ export class InfoAction {
 
     // CDDL
     // info_action = 6
+    writer.writeStartArray(EMBEDDED_GROUP_SIZE);
```
Comment (AngelCastilloB) [inline packages/core/src/Serialization/TransactionBody/ProposalProcedure/InfoAction.ts:27]: Great job @iccicci 🕵🏼 . I think we should extract this change (and the unit test update) to a separate commit as a fix. 

Also we should probably revert the changes in this commit:

https://github.com/input-output-hk/cardano-js-sdk/commit/8ec646a657df80bd0d43f2ea134cffda06e2a4b0

That commit was attempting to fix this issue but was totally wrong 😅. Your change is the correct fix

**Review by AngelCastilloB — CHANGES_REQUESTED** [review]

File: packages/e2e/test/wallet/PersonalWallet/conwayTransactions.test.ts [hunk 5]
```diff
@@ -0,0 +1,395 @@
...
+    }
+  };
+
+  const getDRepCredential = async () => {
+    const drepPubKey = await dRepWallet.getPubDRepKey();
+    const dRepKeyHash = Crypto.Hash28ByteBase16.fromEd25519KeyHashHex(
+      (await Crypto.Ed25519PublicKey.fromHex(drepPubKey).hash()).hex()
+    );
+
+    dRepCredential = { hash: dRepKeyHash, type: CredentialType.KeyHash };
```
Comment (mirceahasegan) [inline packages/e2e/test/wallet/PersonalWallet/conwayTransactions.test.ts:84]: Just nitpicking, but I would prefer to have these get* functions return the values, rather than updating an upper scope variable

Comment (mirceahasegan) [reply packages/e2e/test/wallet/PersonalWallet/conwayTransactions.test.ts:84]: Or better, rename them to `set*`. They are only used in the before* statements to initialize state, so it would make sense to be setters.

File: packages/e2e/test/wallet/PersonalWallet/conwayTransactions.test.ts [hunk 6]
```diff
@@ -0,0 +1,395 @@
...
+  });
+
+  describe('with proposal procedure', () => {
+    let actionId: Cardano.GovernanceActionId;
+
+    beforeAll(async () => {
+      const proposalProcedure: Cardano.ProposalProcedure = {
+        anchor,
+        deposit: governanceActionDeposit,
+        governanceAction: { __typename: GovernanceActionType.info_action },
```
Comment (mirceahasegan) [inline packages/e2e/test/wallet/PersonalWallet/conwayTransactions.test.ts:363]: Do we plan on covering submitting the rest of the actions types?

Comment (iccicci) [reply packages/e2e/test/wallet/PersonalWallet/conwayTransactions.test.ts:363]: Many of them require a Constitutional Commette or an already approved Proposal Procedure... maybe we can add them later, when we have the required scenario...
On the fly I would say we can add only Parameter Change Proposal... let me take a look

**Review by mirceahasegan — APPROVED** [review]
Great work, @iccicci ! 🥇
